scripts/download-engine.cts

```diff
@@ -13,10 +13,16 @@ function downloadEngine() {
     fs.rmSync(engineDir, { recursive: true, force: true });
   }
   fs.mkdirSync(engineDir);
+  downloadEngineReq(engineDir, engineDonwloadUrl);
+}
 
-  const filename = path.join(engineDir, path.basename(engineDonwloadUrl));
-
-  https.get(engineDonwloadUrl, res => {
+function downloadEngineReq(engineDir: string, downloadUrl: string) {
+  const filename = path.join(engineDir, path.basename(downloadUrl));
+  https.get(downloadUrl, res => {
+    if (res.statusCode === 302 && res.headers.location) {
+      downloadEngineReq(engineDir, res.headers.location);
+      return;
+    }
```

Comment:
normally this clients have a follow redirect mode and you don't have to do this on your own...

Reply:
could indeed be more elegant, but that's not the focus for me, just an help script